0000-catalog.md

```diff
@@ -0,0 +1,801 @@
+(Organizations and agents are defined by the Pike smart contract.)
+
+```
+message CatalogUpdateAction {
+    // GS1 Company Prefix from owner + catalog_id are use as
+    // a composite key for determining the state address
+    string owner = 1;
+    string catalog_id = 2;
+    string name = 3;
+    repeated PropertyValues properties = 4;
+}
+```
+
+Validation requirements:
+
+- If a catalog with catalog_id does not exist, the transaction is invalid.
+- The signer of the transaction must be an agent in the Pike state and must
+  belong to an organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_update_catalog for the organization,
+  otherwise the transaction is invalid.
+
+The inputs for CatalogUpdateAction must include:
+
+- Address of the Agent submitting the transaction
+- Address of the Organization the Catalog is being created for
+- Address of the Catalog to be updated
+
+The outputs for CatalogUpdateAction must include:
+
+Address of the Catalog to be updated
+
+### CatalogDeleteAction
+
+CatalogDeleteAction deletes an existing catalog from state. The transaction should be
+submitted by an agent, which is identified by its signing key, acting on behalf 
+of the organization that corresponds to the owner in the delete transaction.
+(Organizations and agents are defined by the Pike smart contract.)
+
+```
+message CatalogDeleteAction {
+    // GS1 Company Prefix from owner + catalog_id are use as
+    // a composite key for determining the state address
+    string owner = 1;
+    string catalog_id = 2;
+}
+```
+
+Validation requirements:
+
+- If a catalog with catalog_id exists the transaction is valid, otherwise it's
+  invalid.
+- The signer of the transaction must be an agent in the Pike state and must
+  belong to an organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_delete_catalog for the organization,
+  otherwise the transaction is invalid.
+
+The inputs for CatalogDeleteAction must include:
+
+- Address of the Agent submitting the transaction
+- Address of the Organization the Catalog is being created for
+- Address of the Catalog to be deleted
+
+The outputs for CatalogDeleteAction must include:
+
+Address of the Catalog to be deleted
+
+**_NOTE: Deleting a catalog is potentially dangerous operation that could leave 
+dangling references and should be done with care._**
+
+## Catalog_Product Actions
+
+### CatalogProductCreateAction
+
+The CatalogProductCreateAction adds a new catalog_product to state. The 
+catalog_product references a Grid Product for the shared item level master 
+data. The transaction should be submitted by an agent, which is identified by 
+its signing key, acting on behalf of the organization that corresponds to the
+owner in the create transaction. (Organizations and agents are defined by the 
+Pike smart contract.)
+
+```
+message CatalogProductCreateAction {
```

Comment:
So does this create a CatalogProduct object in state, or does it update the Grid Product with the given properties?

Reply:
It creates a separate object in state known as `catalog_product`. I look at grid product as the item master data about a particular product. The catalog_product derives from that and adds any attributes that are business partner-specific. 

Return_policy is a good example. The Grid Product would likely not have a return_policy field, if it did it would likely be overridden. The catalog_product would contain the return_policy with the specific business partner relationship policy. Does that make sense?

Reply:
Okay, yes that makes sense to me now. Thanks!